libass/ass_font.c

```diff
@@ -319,6 +319,23 @@ int ass_face_get_weight(FT_Face face)
         return 300 * !!(face->style_flags & FT_STYLE_FLAG_BOLD) + 400;
 }
 
+/**
+ * \brief Get face width
+ **/
+int ass_face_get_width(FT_Face face)
+{
+#if FREETYPE_MAJOR > 2 || (FREETYPE_MAJOR == 2 && FREETYPE_MINOR >= 6)
+    TT_OS2 *os2 = FT_Get_Sfnt_Table(face, FT_SFNT_OS2);
+#else
+    // This old name is still included (as a macro), but deprecated as of 2.6, so avoid using it if we can
+    TT_OS2 *os2 = FT_Get_Sfnt_Table(face, ft_sfnt_os2);
+#endif
+    if (os2 && os2->version != 0xffff && os2->usWidthClass)
+        return os2->usWidthClass;
```

Comment:
This just copies the value in directly, but the value this is stored in is in fontconfig's units, defined in terms of the os2 value here: https://github.com/behdad/fontconfig/blob/5b41ded2b0ddb98a07ac86264b94403cb7a0fd82/src/fcfreetype.c#L1782
And also multiplied by a `wdth` value that comes from… somewhere? I dunno: https://github.com/behdad/fontconfig/blob/5b41ded2b0ddb98a07ac86264b94403cb7a0fd82/src/fcfreetype.c#L1362

Reply:
I should have read the [OS/2 table specs](https://docs.microsoft.com/en-us/typography/opentype/spec/os2#uswidthclass) a bit more closely. The wdth value comes from the fvar table and STAT table it seems?

Reply:
I should have looked at the [OS/2 metrics table spec](https://docs.microsoft.com/en-us/typography/opentype/spec/os2#uswidthclass) more closely. Looks like `wdth` comes from the fvar and STAT tables.